Improved: Convert InvoicePerShipmentTests.xml tests to groovy (OFBIZ-11553)

priyasharma1 opened the pull request:
(OFBIZ-11553)

Done following:
- converted all XML test implementation to groovy
- used MockHttpServletRequest MockHttpServletResponse objects for triggering the events
- removed the XML version of the test implementations
- updated the test service mapping in the invoicetests.xml

Explanation: Since all the test-cases were similar we have combined OFBIZ-11554, OFBIZ-11555, OFBIZ-11556 also.

priyasharma01 committed:
Improved: Convert InvoicePerShipmentTests.xml tests to groovy (OFBIZ-11553)
Done following:
- converted all xml test implementation to groovy
- used MockHttpServletRequest MockHttpServletResponse objects for triggering the events
- removed the xml version of the test implementations
- updated the test service mapping in the invoicetests.xml
(OFBIZ-11553)

JacquesLeRoux commented:
Hi Priya, Pawan,

I suggest we factorise the common part in one specific method, TIA

priyasharma01 committed:
Improved: Moved the common code into a method so as to avoid duplication of code.
- passed in two parameters i.e. product id and the invoicePerShipment flag, to be used as per the need of the testcase.
(OFBIz-11553)

priyasharma1 commented:
Hi @JacquesLeRoux 
I moved the duplicate code to a method and triggered it in the tests as per the conditions.

But the SonarCloud results are yet not good, any other suggestion?

JacquesLeRoux commented:
Hi Priya,

It's good now, the duplicated lines reported are only in jquery. I'll review later, thanks!

priyasharma01 committed:
Improved: Added support to get access to the Debug log methods in the groovy tests. (OFBIZ-11553)
- Also updated the references in the InvoicePerShipment tests.

priyasharma01 committed:
Merge branch 'trunk' of https://github.com/apache/ofbiz-framework into OFBIZ-11553
